docs/topics/exporters.rst

```diff
@@ -36,38 +36,36 @@ to export
 3. and finally call the :meth:`~BaseItemExporter.finish_exporting` to signal
 the end of the exporting process
 
-Here you can see an :doc:`Item Pipeline <item-pipeline>` which uses an Item
-Exporter to export scraped items to different files, one per spider::
-
-   from scrapy import signals
-   from scrapy.exporters import XmlItemExporter
-
-   class XmlExportPipeline(object):
-
-       def __init__(self):
-           self.files = {}
-
-        @classmethod
-        def from_crawler(cls, crawler):
-            pipeline = cls()
-            crawler.signals.connect(pipeline.spider_opened, signals.spider_opened)
-            crawler.signals.connect(pipeline.spider_closed, signals.spider_closed)
-            return pipeline
-
-       def spider_opened(self, spider):
-           file = open('%s_products.xml' % spider.name, 'w+b')
-           self.files[spider] = file
-           self.exporter = XmlItemExporter(file)
-           self.exporter.start_exporting()
-
-       def spider_closed(self, spider):
-           self.exporter.finish_exporting()
-           file = self.files.pop(spider)
-           file.close()
-
-       def process_item(self, item, spider):
-           self.exporter.export_item(item)
-           return item
+Here you can see an :doc:`Item Pipeline <item-pipeline>` which uses multiple
+Item Exporters to group scraped items to different files according to the
+value of one of their fields::
+
+    from scrapy.exporters import XmlItemExporter
+
+    class PerYearXmlExportPipeline(object):
+        """Distribute items across multiple XML files according to their 'year' field"""
+
+        def open_spider(self, spider):
+            self.year_to_exporter = {}
+
+        def close_spider(self, spider):
+            for exporter in self.year_to_exporter.itervalues():
```

Comment:
It should be .values() in order to be compatible with Python 3. Creating a temporary list in Python 2 is not a problem as a list would be small.